components/collector/tests/source_collectors/dependency_track/test_source_up_to_dateness.py

```diff
@@ -20,7 +20,7 @@ class DependencyTrackSourceUpToDatenessVersionTest(DependencyTrackTestCase):
     METRIC_TYPE = "source_up_to_dateness"
     LANDING_URL = "https://dependency_track"
 
-    def projects(self) -> list[DependencyTrackProject]:
+    def projects(self, version: str = "1.4") -> list[DependencyTrackProject]:
```

Comment:
Do we test with version not being given?

Reply:
Yes:

```python
    async def test_filter_by_project_version_when_project_has_no_version(self):
        """Test filtering projects by version."""
        self.set_source_parameter("project_versions", ["1.2", "1.3"])
        response = await self.collect(get_request_json_return_value=self.projects(version=""))
        self.assert_measurement(response, value="0", entities=[])
```